src/dparse/lexer.d

```diff
@@ -1851,36 +1851,43 @@ in
 }
 do
 {
-    import std.string : lineSplitter, stripRight;
+    import std.string : lineSplitter, stripRight, KeepTerminator;
 
     string leadingChars;
-    size_t i = 3;
-    size_t j;
-    bool hasOutput = false;
-    bool lastWasBlank = false;
+
+	enum LineType { none, normal, strange }
+	LineType prevLineType;
+
     switch (comment[0 .. 3])
     {
     case "///":
-        j = comment.length;
-
-        foreach (line; lineSplitter(comment))
-        {
+        foreach (line; lineSplitter!(KeepTerminator.yes)(comment))
+        {
+			immutable LineType currentLineType = line.endsWith("\n")
+					|| line.endsWith("\r")
+					|| line.endsWith("\u2029")
+					|| line.endsWith("\u2030")
```

Comment:
you used paragraph separater and permille sign here, should be `\u2028` and `\u2029`